### ISSUE TITLE
I get different results depending on how I run the tool

### ISSUE THREAD

**Issue by greenc05** [bbc/bbc-a11y#165]
## Summary
I get different results depending on how I run the tool. Should I run it the following 3 ways to make sure all errors are picked up??

Also when running the tool different ways not only to I get a different number of errors, I get different types of errors.

Also some of the errors are not actually errors, for example when running the tool the 1st way below, neither of the errors are actually errors, the page has a html lang attribute and a main landmark.

## Current Behavior
Testing url: http://www.test.bbc.co.uk/front_page_beta

**When running this command a11y http://www.test.bbc.co.uk/front_page_beta
I get the following results:**

✗ http://www.test.bbc.co.uk/front_page_beta
  * Indicating Language: Html must have lang attribute
    - html tag has no lang attribute: /html
  * Main Landmark: Exactly one main landmark
    - Found 0 elements with role="main".

1 page checked, 2 errors found, 0 standards skipped

**When running this command ./node_modules/.bin/bbc-a11y bbc-a11y with page('http://www.test.bbc.co.uk/news/front_page_beta') in the a11y.js config file I get the following results:**

bbc-a11y
  * Headings: Exactly one main heading
    - Found 0 h1 elements.
  * Indicating Language: Html must have lang attribute
    - html tag has no lang attribute: /html
  * Main Landmark: Exactly one main landmark
    - Found 0 elements with role="main".

1 page checked, 3 errors found, 0 standards skipped

**When running this command ./node_modules/.bin/bbc-a11y http://www.test.bbc.co.uk/news/front_page_beta --interactive I get the following errors:**

✗ http://www.test.bbc.co.uk/news/front_page_beta
  * Form Interactions: Forms must have submit buttons
    - Form has no submit button: //form[@id='orb-search-form']
  * Headings: Content must follow headings
    - No content follows: //div[@id='u03272949671372771']/div/div/div/div[3]/div/div[2]/div/div/h3
  * Headings: Exactly one main heading
    - Found 0 h1 elements.
  * Image Alternatives: Images must have alt attributes
    - Image has no alt attribute: //div[@id='u03272949671372771']/div/div/div/div[3]/div/div[1]/div/div/img
  * Minimum Text Size: Text cannot be too small
    - Text size too small (10px): //a[@id='brand']/span[2]
    - Text size too small (10px): //div[@id='u15665784594602883']/div/div/div/div[1]/div/div[1]/div/div/div/div[3]/div/div/h4

1 page checked, 6 errors found, 0 standards skipped

**Comment by joshski** [comment]
Thanks @greenc05 

The first example you posted looks like you're running the Ruby version of bbc-a11y (which you run with the command `a11y` and not `bbc-a11y`) so it's not surprising that you're seeing different results.

I tried to reproduce the discrepancy between the other two and I'm not seeing the same thing. The URL you mentioned is responding with a 404 page at the moment (at least from where I am) which produces the following output, whether I call bbc-a11y with a URL or a page in the config file: 

```
➜  node_modules/.bin/bbc-a11y http://www.test.bbc.co.uk/front_page_beta
✗ http://www.test.bbc.co.uk/front_page_beta
  * Indicating Language: Html must have lang attribute
    - html tag has no lang attribute: /html
  * Main Landmark: Exactly one main landmark
    - Found 0 elements with role="main".

1 page checked, 2 errors found, 0 standards skipped

➜  echo "page('http://www.test.bbc.co.uk/front_page_beta')" > a11y.js
➜  node_modules/.bin/bbc-a11y                                                        
✗ http://www.test.bbc.co.uk/front_page_beta
  * Indicating Language: Html must have lang attribute
    - html tag has no lang attribute: /html
  * Main Landmark: Exactly one main landmark
    - Found 0 elements with role="main".

1 page checked, 2 errors found, 0 standards skipped                                                                           ```

**Comment by joshski** [comment]
Is there any chance the URL you mentioned was responding with different content between each test run?

**Comment by joshski** [comment]
Here's what I saw on that URL when I ran the tests above:
<img width="467" alt="screen shot 2017-01-26 at 18 39 03" src="https://cloud.githubusercontent.com/assets/22017/22345414/e4997a48-e3f7-11e6-9b1b-ae4f3c2f588b.png">

**Comment by greenc05** [comment]
@joshski hummm not sure, that url should work, can you see test.bbc.co.uk? I ran all the tests at the same time, so the content wouldn't of been greatly different if at all. 

Just trying again now, so running the url from the config file and in the interactive mode gives me the same result, if I run node_modules/.bin/bbc-a11y http://www.test.bbc.co.uk/front_page_beta then it gets the same result as you, should that work too?

**Comment by joshski** [comment]
I still can't see any content at that URL.

All of these commands should return exactly the same result:

`node_modules/.bin/bbc-a11y http://www.test.bbc.co.uk/front_page_beta`

`node_modules/.bin/bbc-a11y http://www.test.bbc.co.uk/front_page_beta --interactive`

`echo "page('http://www.test.bbc.co.uk/front_page_beta')" > a11y.js && node_modules/.bin/bbc-a11y`

If that's still not the case for you, I'm going to need to get inside your network to work out why :)

**Comment by greenc05** [comment]
Ok I've wiped all my bbc-a11y code and started a fresh @joshski, now when running the following commands I get the same results (though one of my colleagues gets one extra error when we ran the tool at the same time)

bbc-a11y http://www.test.bbc.co.uk/news/front_page_beta
bbc-a11y http://www.test.bbc.co.uk/news/front_page_beta --interactive
bbc-a11y (with the same url in the config file)

Out of interest, if I don't specify a width for the viewport, what is the default size it looks at? It would be nice to add this information to the readme

I also assume the tool checks the url as if it has never been viewed before? e.g. without any location settings or any other assumed knowledge

**Comment by joshski** [comment]
Thanks @greenc05 

> Out of interest, if I don't specify a width for the viewport, what is the default size it looks at? It would be nice to add this information to the readme

The default width is 1024px and I've documented that in the readme now. Actually your comment made me realise there was a bug with viewport width, when one page specifies it and the next doesn't, the default is not restored. That's now fixed in master and will be released today. Thanks for prompting it!

So I'm closing this one now , but I've added an issue for your final question.
